src/pages/Autolending/AutolendingWhen.vue

```diff
@@ -140,20 +146,30 @@ export default {
 					lendAfterDaysIdle
 				}
 			}
+			my {
+				autoDeposit {
+					donateAmount
+				}
+			}
 		}`,
 		preFetch: true,
 		result({ data }) {
 			this.isEnabled = !!_get(data, 'autolending.currentProfile.isEnabled');
-			const donationPercentage = _get(data, 'autolending.currentProfile.donationPercentage');
 			this.lendAfterDaysIdle = _get(data, 'autolending.currentProfile.lendAfterDaysIdle');
 			this.isChanged = !!_get(data, 'autolending.profileChanged');
+			const donationPercentage = _get(data, 'autolending.currentProfile.donationPercentage');
 			this.donation = _isFinite(donationPercentage) ? donationPercentage : 15;
 			this.donationToggle = this.donation !== 0 ? 'on' : 'off';
+			this.hasAutoDepositDonation = _get(data, 'my.autoDeposit.donateAmount') > 0;
 		},
 	},
 	mounted() {
 		// After initial value is loaded, setup watch
 		this.$watch('donation', (donation, previousDonation) => {
+			if (this.hasAutoDepositDonation) {
```

Comment:
If I understand correctly, the donation is getting zeroed out on the backend? If that is the case, we shouldn't need this code here. I would also prefer to avoid the param reassign.

Reply:
Fair point. Reverted.